src/Features/Core/Portable/EditAndContinue/EditSession.cs

```diff
@@ -364,6 +517,135 @@ internal ImmutableArray<DocumentId> GetDocumentsWithReportedRudeEdits()
             }
         }
 
+        // internal for testing
+        internal static void GetActiveStatementAndExceptionRegionSpans(
```

Comment:
>GetActiveStatementAndExceptionRegionSpans [](start = 29, length = 41)

It looks like O(N^2). Hopefully used just for testing.

Reply:
No, this is product code. It's O(M*M) where N is number of updated active statements and M is number of updated methods. Both are usually small, so I don't think that's an issue. We can optimize if we find it to be an issue.

---
In reply to: [162478590](https://github.com/dotnet/roslyn/pull/24316#discussion_r162478590) [](ancestors = 162478590)